test/unit/forge/routes/api/teamDevices_spec.js

```diff
@@ -116,7 +115,199 @@ describe('Team Devices API', function () {
+                devices.should.have.length(1)
+            })
+
+            it('by type', async function () {
+                // Match device on type
+                const devices = await queryDevices(`/api/v1/teams/${TestObjects.ATeam.hashid}/devices?query=another%20type`)
+                devices.should.have.length(2)
+            })
+        })
+
+        describe('Supports sorting the devices', function () {
+            before(async function () {
+                const application2 = await app.factory.createApplication({ name: 'application-2' }, TestObjects.ATeam)
+
+                const instance2 = await app.factory.createInstance(
+                    { name: 'instance-2' },
+                    application2,
+                    app.stack,
+                    app.template,
+                    app.projectType,
+                    { start: false }
+                )
+
+                // Add 3 more devices
+                await app.factory.createDevice({ name: 'device 2' }, TestObjects.ATeam, TestObjects.Project1)
+                await app.factory.createDevice({ name: 'device 3', type: 'it is another type of device' }, TestObjects.ATeam, instance2)
+                await app.factory.createDevice({ name: 'device 4', type: 'it is another type of device, no instance' }, TestObjects.ATeam)
+            })
+
+            after(async function () {
+                await app.db.models.Device.destroy({
+                    where: {
+                        name: ['device 2', 'device 3', 'device 4']
+                    }
+                })
+            })
+
+            it('by name', async function () {
+                // Sort by name ASC (default)
+                const resultName = await queryDevices(`/api/v1/teams/${TestObjects.ATeam.hashid}/devices?sort=name`)
+                resultName.map((device) => device.name).should.match(['device 1', 'device 2', 'device 3', 'device 4'])
+
+                // Sort by name DESC (explicit)
+                const resultNameDesc = await queryDevices(`/api/v1/teams/${TestObjects.ATeam.hashid}/devices?sort=name&order=desc`)
+                resultNameDesc.map((device) => device.name).should.match(['device 4', 'device 3', 'device 2', 'device 1'])
+
+                // Sort by name ASC (explicit)
+                const resultNameAsc = await queryDevices(`/api/v1/teams/${TestObjects.ATeam.hashid}/devices?sort=name&order=asc`)
+                resultNameAsc.map((device) => device.name).should.match(['device 1', 'device 2', 'device 3', 'device 4'])
+            })
+
+            it('by instance->application name', async function () {
+                // Sort by application name ASC (default)
+                const resultName = await queryDevices(`/api/v1/teams/${TestObjects.ATeam.hashid}/devices?sort=application`)
+
+                // SQLite puts undefined includes first ASC, Postgres has it last...
+                const usingSQLite = resultName[0].application === undefined
```

Comment:
This looks a little brittle to me. Why not check the database type instead?  `getDialect()` perhaps?